src/sentry/taskworker/registry.py

```diff
@@ -126,13 +132,16 @@ def wrapped(func: Callable[P, R]) -> Task[P, R]:
 
         return wrapped
 
-    def send_task(self, activation: TaskActivation) -> None:
+    def send_task(self, activation: TaskActivation, wait_for_delivery: bool = False) -> None:
         metrics.incr("taskworker.registry.send_task", tags={"namespace": activation.namespace})
-        # TODO(taskworker) producer callback handling
-        self.producer.produce(
+
+        produce_future = self.producer.produce(
             ArroyoTopic(name=self.topic.value),
             KafkaPayload(key=None, value=activation.SerializeToString(), headers=[]),
         )
+        if wait_for_delivery:
+            while not produce_future.is_done():
+                time.sleep(0.1)
```

Comment:
I can't seem to find `is_done()` in the docs for `concurrent.futures`, it might be better to use `result(timeout=None)` here
```suggestion
            produce_future.result(timeout=None)
```

Reply:
There is a [done()](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Future.done) on `Future`, but I'm not seeing `is_done()` either.